dev/Code/CryEngine/CryPhysics/geoman.cpp

```diff
@@ -183,7 +183,13 @@ int CGeomManager::AddRefGeometry(phys_geometry* pgeom)
 
 int CGeomManager::UnregisterGeometry(phys_geometry* pgeom)
 {
-    WriteLock lock(m_lockGeoman);
+	WriteLock lock(m_lockGeoman);
+	return UnregisterGeometryLocked(pgeom);
+}
+
+// Internal function that assumes that the caller is already in the write lock, see ::UnregisterGeometry
+int CGeomManager::UnregisterGeometryLocked(phys_geometry* pgeom)      
```

Comment:
Added internal function that assumes that the caller is already in the write lock